feat(pagination): add disabled variation

christiemolloy opened the pull request:
closes #1963 

Adds disabled variation for the top pagination component - assuming that we don't need to also show an example for the bottom variation?

patternfly-build commented:
Deploy preview for *pf-next* ready!

Built with commit 231f6c7119a4878085004dad03d0232064189eb5

https://deploy-preview-2015--pf-next.netlify.com

christiemolloy commented:
Thanks @mattnolting updated!!

mcoker commented:
@christiemolloy looks like in https://github.com/patternfly/patternfly-next/pull/1973, we missed adding `disabled` to the toggle. Here's the discussion on it when we added the disabled variation of the dropdown component - https://github.com/patternfly/patternfly-next/pull/1512#pullrequestreview-210241111 and you'll find it on the dropdown's disabled example - https://pf4.patternfly.org/components/Dropdown/examples/.

And in this PR, we need to add `disabled` to `.pf-c-options-menu__toggle-button`, which is covered in the docs above, which are also here in the options menu docs

> When [`.pf-m-disabled`] is used, disabled should also be added to any form elements in .pf-c-options-menu__toggle

mcoker commented on src/patternfly/components/Pagination/examples/pagination-top-disabled-example.hbs:
```diff
@@ -0,0 +1,28 @@
+  {{> pagination-options-menu id="pagination-options-menu-top-disabled-example"  options-menu--IsText="true" options-menu-toggle--IsDisabled="true"}}
+
+  {{#> pagination-nav}}
+    {{#> button button--modifier="pf-m-plain pf-m-disabled" button--attribute='aria-label="Go to first page" aria-disabled="true"'}}
```
Looks like in the pagination component, we're using `pf-m-disabled` wrong. We should be using `disabled` on `button`s and `pf-m-disabled` on `a` elements. So these buttons need to be updated to use `disabled`

Do you want to do that as part of this issue, or a follow up issue?

christiemolloy replied:
I can do it in this issue

christiemolloy commented:
@mcoker updated... Added disabled variation to the plain options menu with text. I also added `disabled` to both `pf-c-options-menu__toggle` and `pf-c-options-menu__toggle-button` .. In the dropdown component I don't see the `pf-m-disabled` modifier on the buttons, I only see `disabled`, so i'm a little confused as to what to do there.

<img width="477" alt="Screen Shot 2019-07-08 at 10 45 44 AM" src="https://user-images.githubusercontent.com/20118816/60819471-a5ff7400-a16d-11e9-86b9-5f8242ecb71d.png">

christiemolloy commented on src/patternfly/components/OptionsMenu/docs/code.md:
```diff
@@ -6,7 +6,7 @@
 | -- | -- | -- |
```
@mcoker @jgiardino now that the react implementation is complete should I create an issue to fill in the options menu docs? or do you know if this is still on hold?

mcoker commented on src/patternfly/components/Pagination/examples/pagination-bottom-example.hbs:
```diff
@@ -2,10 +2,10 @@
 
   {{#> pagination-nav}}
-    {{#> button button--modifier="pf-m-plain pf-m-disabled" button--attribute='aria-label="Go to first page" aria-disabled="true"'}}
+    {{#> button button--modifier="pf-m-plain" button--attribute='disabled aria-label="Go to first page" aria-disabled="true"'}}
```
I don't think we need `aria-disabled`, either, if the element has the `disabled` attribute. @jgiardino can you confirm?

christiemolloy replied:
In this article it says: "A button with disabled as well as the button with aria-disabled will be labeled as "dimmed button" https://a11y-101.com/development/aria-disabled

christiemolloy replied:
Looks like we only need "disabled"

mcoker commented:
On the "Options menu - disabled" and "Options menu - plain
" (disabled) examples, we don't need `.pf-m-disabled` on the toggle since it's a `<button>`.

christiemolloy commented:
thanks @mcoker updated!

mcoker commented on src/patternfly/components/OptionsMenu/options-menu.scss:
```diff
@@ -174,9 +174,9 @@
   &:disabled {
     pointer-events: none;
 
-    &:not(.pf-m-plain) {
```
Removing this gives a background to the plain (non `.pf-m-text`) button.

<img width="186" alt="Screen Shot 2019-07-09 at 10 32 16 AM" src="https://user-images.githubusercontent.com/35148959/60902272-fa215b80-a234-11e9-82a2-8dfeb933947e.png">

I also don't know that you need the `&:not(.pf-m-plain)` on L179, either. It's correct code-wise, but seems unnecessary and increases specificity.

What if you rewrote this block to


```
  &.pf-m-disabled,
  &:disabled {
    pointer-events: none;

    &:not(.pf-m-plain),
    &.pf-m-text {
      --pf-c-options-menu__toggle--Background: var(--pf-c-options-menu__toggle--disabled--BackgroundColor);
    }

    &::before {
      border: 0;
    }
  }
```

christiemolloy commented:
I'm good with that change! updated! @mcoker

mcoker commented on src/patternfly/components/OptionsMenu/docs/code.md:
```diff
@@ -6,7 +6,7 @@
 | `role` or `aria` | `pf-c-options-menu` |  accessibility notes. |
 *Note:* The attribute `aria-selected="true"` should be set programmatically to the selected item(s).
-
+| `disabled` | `.pf-c-options-menu__toggle`, `.pf-c-options-menu__toggle-button` | Disables the options menu toggle and toggle button and removes it from keyboard focus. |
```
I think you want to move this above the `*Note:*`

patternfly-build commented:
:tada: This PR is included in version 2.20.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@patternfly/patternfly)
- [GitHub release](https://github.com/patternfly/patternfly-next/releases/tag/v2.20.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
